[18.0][MIG] stock_picking_report_undelivered_product: Migration to 18.0

juancarlosonate-tecnativa opened the pull request:
Since odoo base in the delivery slip report already generates a table for undelivered products, i try to reuse as much as possible to avoid code duplication. Retrieve all the stock moves based on the applied filter and set them accordingly. Then, perform a small operation to calculate the actual undelivered quantities: bo_line.product_uom_qty - bo_line.quantity

@Tecnativa TT54343

@CarlosRoca13 @sergio-teruel can you check? thanks!

sergio-teruel committed:
[ADD] stock_picking_report_undelivered_product: New module to display undelivered products in delivery slip report

sergio-teruel committed:
[ADD] stock_picking_report_undelivered_product: New module to display undelivered products in delivery slip report

oca-travis committed:
[UPD] Update stock_picking_report_undelivered_product.pot

oca-travis committed:
[UPD] Update stock_picking_report_undelivered_product.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[ADD] icon.png

sergio-teruel committed:
[IMP] stock_picking_report_undelivered_product: black, isort

sergio-teruel committed:
[IMP] stock_picking_report_undelivered_product: black, isort

sergio-teruel committed:
[MIG] stock_picking_report_undelivered_product: Migration to v13.0

sergio-teruel committed:
[MIG] stock_picking_report_undelivered_product: Migration to v13.0

oca-travis committed:
[UPD] Update stock_picking_report_undelivered_product.pot

oca-travis committed:
[UPD] Update stock_picking_report_undelivered_product.pot

sergio-teruel committed:
[MIG] stock_picking_report_undelivered_product: Migration to v15.0

sergio-teruel committed:
[MIG] stock_picking_report_undelivered_product: Migration to v15.0

Commit:
[UPD] Update stock_picking_report_undelivered_product.pot

Commit:
[UPD] Update stock_picking_report_undelivered_product.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-reporting-15.0/stock-logistics-reporting-15.0-stock_picking_report_undelivered_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-reporting-15-0/stock-logistics-reporting-15-0-stock_picking_report_undelivered_product/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-reporting-15.0/stock-logistics-reporting-15.0-stock_picking_report_undelivered_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-reporting-15-0/stock-logistics-reporting-15-0-stock_picking_report_undelivered_product/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (19 of 19 strings)

Translation: stock-logistics-reporting-15.0/stock-logistics-reporting-15.0-stock_picking_report_undelivered_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-reporting-15-0/stock-logistics-reporting-15-0-stock_picking_report_undelivered_product/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (19 of 19 strings)

Translation: stock-logistics-reporting-15.0/stock-logistics-reporting-15.0-stock_picking_report_undelivered_product
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-reporting-15-0/stock-logistics-reporting-15-0-stock_picking_report_undelivered_product/es/

juancarlosonate-tecnativa committed:
[IMP] stock_picking_report_undelivered_product: pre-commit auto fixes

juancarlosonate-tecnativa committed:
[MIG] stock_picking_report_undelivered_product: Migration to 18.0

TT54343

sergio-teruel commented on stock_picking_report_undelivered_product/views/report_deliveryslip.xml:
```diff
@@ -1,95 +1,56 @@
+        inherit_id="stock.report_delivery_document"
+        priority="200"
+    >
+        <xpath expr="//*[@t-set='backorders']" position="replace">
```
Upps! Try not replace any view!! Hide it instead!

juancarlosonate-tecnativa replied:
Done!

sergio-teruel reviewed (changes requested):
Great @juancarlosonate-tecnativa  !!!

Minor fix!!

sergio-teruel commented:
Review tests!!

rousseldenis commented:
/ocabot migration stock_picking_report_undelivered_product

rousseldenis reviewed (changes requested):
Agree with @sergio-teruel. Don't replace views

CarlosRoca13 reviewed (approved):
Thanks!

pedrobaeza commented:
@rousseldenis

juancarlosonate-tecnativa commented:
@rousseldenis @sergio-teruel check again please!

sergio-teruel commented on stock_picking_report_undelivered_product/views/report_deliveryslip.xml:
```diff
@@ -0,0 +1,63 @@
+
+        <xpath expr="//span[@t-field='bo_line.product_uom_qty']/.." position="after">
+            <td class="text-center w-auto">
+                <span t-esc="bo_line.product_uom_qty - bo_line.quantity">3.00</span>
```
Why 3.0?

sergio-teruel replied:
Upsss I see... it's in original view...
Do you know that you can user attribute 'move' to move xml elements??

sergio-teruel replied:
You can see here https://github.com/odoo/odoo/blob/7de8f1edaa91a987161774d6b86220f58ea3354d/addons/project/views/rating_rating_views.xml#L33

juancarlosonate-tecnativa replied:
i know it exists, @CarlosRoca13  mentioned it to me, but i don't know how to use it. Can you give me some more details?

juancarlosonate-tecnativa replied:
please check again

rousseldenis commented on stock_picking_report_undelivered_product/models/stock_move.py:
```diff
@@ -1,20 +0,0 @@
-class StockMove(models.Model):
-    _inherit = "stock.move"
-
-    splitted_stock_move_orig_id = fields.Many2one(
```
@juancarlosonate-tecnativa Do we lose the information on existing databases ?

juancarlosonate-tecnativa replied:
this information is no longer useful and will never be used

sergio-teruel replied:
Maybe this change needs to be analysed in more depth.

pedrobaeza commented:
@carlosdauden @juancarlosonate-tecnativa can you check this?

RobbeDynapps commented:
@carlosdauden @juancarlosonate-tecnativa Can you check again?

carlosdauden commented:
Can you rebase to force runboat build?

juancarlosonate-tecnativa commented:
> Can you rebase to force runboat build?

Done

carlosdauden commented:
I'm testing on Runboat with a delivery order that contains two products: one is partially delivered, and the other is not delivered at all. I validate without creating a backorder, and when I print, it doesn't matter which option I configure in the settings — it always shows all the quantities still to be delivered.
http://oca-stock-logistics-reporting-18-0-pr367-7cf7683e2915.runboat.odoo-community.org/odoo/sales/25/action-365/24?debug=1

juancarlosonate-tecnativa commented:
> I'm testing on Runboat with a delivery order that contains two products: one is partially delivered, and the other is not delivered at all. I validate without creating a backorder, and when I print, it doesn't matter which option I configure in the settings — it always shows all the quantities still to be delivered. http://oca-stock-logistics-reporting-18-0-pr367-7cf7683e2915.runboat.odoo-community.org/odoo/sales/25/action-365/24?debug=1

@carlosdauden Yes, and in principle it’s correct. Even if you don’t create a backorder, the undelivered quantities are still displayed. That’s how it worked in previous versions.

carlosdauden commented:
When the ‘completely_undelivered’ option is selected in the ‘undelivered_product_slip_report_method’ field, the behaviour should change and the partially delivered movement should not be displayed.

@sergio-teruel, does this not work this way in 15.0?

CarlosRoca13 committed:
[FIX] stock_picking_report_undelivered_product: Error when trying to sort products with and without default_code

When a picking has moves with products that has default_code and other moves that hasn't it. The system throws an error when trying to print the Delivery Note and they are not delivered yet.

OCA-git-bot committed:
[BOT] post-merge updates

juancarlosonate-tecnativa committed:
[IMP] stock_picking_report_undelivered_product: pre-commit auto fixes

juancarlosonate-tecnativa committed:
[MIG] stock_picking_report_undelivered_product: Migration to 18.0

TT54343

juancarlosonate-tecnativa commented:
Some changes were needed to make the migration work properly. Please review again, thanks!
